pkg/wireguard/agent/agent.go

```diff
@@ -397,33 +455,82 @@ func (a *Agent) UpdatePeer(nodeName, pubKeyHex string, nodeIPv4, nodeIPv6 net.IP
 		return fmt.Errorf("failed to resolve peer endpoint address: %w", err)
 	}
 
-	peer := &peerConfig{
+        peer := &peerConfig{
 		pubKey:     pubKey,
 		endpoint:   epAddr,
 		nodeIPv4:   nodeIPv4,
 		nodeIPv6:   nodeIPv6,
 		allowedIPs: allowedIPs,
 	}
 
+	a.peerByNodeName[nodeName] = peer
+        a.nodeNameByPubKey[pubKey] = nodeName
+        if nodeIPv4 != nil {
+                a.nodeNameByNodeIP[nodeIPv4.String()] = nodeName
+        }
+        if nodeIPv6 != nil {
+                a.nodeNameByNodeIP[nodeIPv6.String()] = nodeName
+        }
+
+        err = a.reassemblyMasterHubConfig()
+
+	if err != nil {
+		return fmt.Errorf("failed to reasmbely master hub configs: %w", err)
+	}
+	// Now here configure the master (hub) peer config
+	masterNodeFullname := "kubernetes/" + a.masterNodeName
+	fmt.Println("++++++++configure master node peerconfig:")
+	if peerMaster := a.peerByNodeName[masterNodeFullname]; peerMaster != nil && masterNodeFullname == nodeName {
+		masterNodeInternalIP := peerMaster.nodeIPv4.String() + ":51871"
+		fmt.Println("+++++++++ masternode internal IP:", masterNodeInternalIP)
+		masterepAddr, _ := net.ResolveUDPAddr("udp", masterNodeInternalIP)
+		a.peerMasternode.pubKey = peerMaster.pubKey
+		a.peerMasternode.endpoint = masterepAddr
+		a.peerMasternode.nodeIPv4 = peerMaster.nodeIPv4
+		a.peerMasternode.nodeIPv6 = peerMaster.nodeIPv6
```

Comment:
same here, i think `peerMasternode` needs to be set to non-hub peer but hub node?

Reply:
peerMasterNode 's target is to keep and store the master node(as hub)'s configuration information here. it should be set to hub node